chain/client/src/stateless_validation/chunk_validator/mod.rs

```diff
@@ -632,6 +632,24 @@ impl Client {
         // wait for validation to finish.
         self.send_state_witness_ack(&witness);
 
+        // Avoid validating state witness for old chunks.
+        // In particular it is impossible for a chunk created at a height
+        // that doesn't exceed the height of the current final block to be
+        // included in the chain. This addresses both network-delayed messages
+        // as well as malicious behavior of a chunk producer.
+        if let Ok(final_head) = self.chain.final_head() {
+            if witness.chunk_header.height_created() <= final_head.height {
+                tracing::debug!(
+                    target: "stateless_validation",
+                    chunk_hash=?witness.chunk_header.chunk_hash(),
+                    witness_height=witness.chunk_header.height_created(),
+                    final_height=final_head.height,
+                    "Skipping old state witness",
```

Comment:
maybe expand this string to day skipping old state witness generated for already final blocks?